framework/contracts/native/ibc-host/src/endpoints/packet.rs

```diff
@@ -145,18 +149,36 @@ pub fn handle_host_module_action(
         _ => {}
     }
 
-    // We pass the message on to the module
-    let msg = wasm_execute(
-        target_module_resolved.address,
-        &MiddlewareExecMsg::ModuleIbc::<Empty, Empty>(ModuleIbcMsg {
-            client_chain,
-            source_module: source_module.module_info,
+    let response = Response::new().add_attribute("action", "module-ibc-call");
+    if is_query {
+        // We query module and set data to the response that can be parsed later
+        let query = QueryRequest::<Empty>::from(WasmQuery::Smart {
+            contract_addr: target_module_resolved.address.into_string(),
             msg,
-        }),
-        vec![],
-    )?;
+        });
+        let bin = match deps.querier.raw_query(&to_json_vec(&query)?) {
+            SystemResult::Err(system_err) => Err(StdError::generic_err(format!(
+                "Querier system error: {system_err}"
+            ))),
+            SystemResult::Ok(ContractResult::Err(contract_err)) => Err(StdError::generic_err(
+                format!("Querier contract error: {contract_err}"),
+            )),
+            SystemResult::Ok(ContractResult::Ok(value)) => Ok(value),
+        }?;
+
+        Ok(response.set_data(bin))
```

Comment:
What's the lifecycle of this bin data? It's catched by polytone and added to the execute response?

Reply:
Polytone collect all responses, so it will return as ibc-host execution response with this data